### DISCUSSION TITLE
关于部分章节标题设置段前段后间距和ABSTRACT的单独设置字体

### DISCUSSION THREAD

**Discussion by heavies-05-bidets** [sjtug/SJTUThesis#970] [提问] [answered]
图书馆专家意见：
3.请确认“ABSTRACT”字体是否为Arial 16磅加粗；

4.请确认各个章节标题（包括摘要、目录、致谢、参考文献、英文大摘要等）段前24磅，段后18磅

请问一下怎么单独给ABSTRACT设置字体和大小、加粗呢？（cls实在看不懂）

各个章节标题（包括摘要、目录、致谢、参考文献、英文大摘要等）设置段前段后距离怎么做呢？

**Comment by LogCreative** [comment, marked as answer]
如果"ABSTRACT"字体必须是 Arial 的话：
```latex
{
\newfontface{\arial}{Arial}[Scale=0.94]
\ctexset{chapter/format+={\arial}}
\begin{abstract}[en]
  % 英文摘要...
\end{abstract}
}
```

关于行间距问题，因为 Word 模板开了文档网格，其间距本身就是不准确的。如果专家认为需要增加间距，则象征性地添加一些段后距即可：
```latex
{
\ctexset{chapter={afterskip=26bp}}
% 摘要
\input{contents/abstract}
}
```
其他部分同理。

**Reply by heavies-05-bidets** [reply]
章节标题行间距可以按照你所提供的代码调整，但是改Arial字体的代码我试了并没有生效，不知道为什么。字体需要提前导入到LaTeX中吗？

**Reply by LogCreative** [reply]
你可以截图看看效果

**Reply by heavies-05-bidets** [reply]
破案了，论文里面的abstract本来就是arial字体。我之所以觉得不是，是因为给的LaTeX模板里面，pdf参考版本的abstract并不是arial字体🤣

**Reply by frelwx** [reply]
请问在替换原本英文摘要时，`\newfontface{\arial}{Arial}[Scale=0.94]`报错，请问是需要导入额外的包吗？
![image](https://github.com/sjtug/SJTUThesis/assets/57762884/4b8d83a2-8e28-4082-8455-a1a79e8b51db)
.......................
破案了，没用xelatex编译。

**Comment by AlexaraWu** [comment]
> 3. 请确认“ABSTRACT”字体是否为Arial 16磅加粗；

讲个笑话，Word 模板里 ABSTRACT 的字体是 Arial Unicode MS，还不是 Arial。这两字体英文字母不一样。

**Reply by LogCreative** [reply]
确实

<img width="338" alt="截屏2023-06-14 07 36 19" src="https://github.com/sjtug/SJTUThesis/assets/61653082/080f98d3-3579-4884-915f-e799a4c066d9">

**Reply by AlexaraWu** [reply]
还有一点是 Office 2016 及以后的版本都不附带 Arial Unicode MS 字体了，所以 Word 这里会 fallback 成 Calibri(?)。如果图书馆的老师是按照 Calibri 来审阅的话，就更搞笑了。

**Reply by LogCreative** [reply]
来自 Microsoft 365 Word（Windows）：
![image](https://github.com/sjtug/SJTUThesis/assets/61653082/536413fa-1f65-4b65-b383-44a45164ee6e)
当然找不到字体 导出 PDF 会在不同的浏览器中显示不同的字体（

估计这位老师还是会按提前给出的 PDF 版本审核

**Reply by AlexaraWu** [reply]
>导出 PDF 会在不同的浏览器中显示不同的字体

不同软件的 fallback 字体不一样。

**Reply by AlexaraWu** [reply]
<img src="https://github.com/sjtug/SJTUThesis/assets/15205102/0db8b27b-2691-492a-9cda-af0fae366acc" alt="fonts-comparison" width=500px>

Arial Unicode MS 只有单字重。Helvetica 类字体和 Arial 主要区别是字母 R 不一样。 

```latex
\documentclass{standalone}

\usepackage{array}
\usepackage{fontspec}

\def\sample{ABSTRACT}

\begin{document}

\LARGE
\begin{tabular}{>{\normalsize}l>{\bfseries}l}
  Arial Unicode MS & {\fontspec[BoldFont=*]{Arial Unicode MS} \sample} \\
  Arial Unicode MS (FakeBold) & {\fontspec[AutoFakeBold=3]{Arial Unicode MS} \sample} \\
  Arial & {\fontspec{Arial} \sample} \\
  Helvetica & {\usefont{T1}{phv}{b}{n} \sample} \\
  TeX Gyre Heros & {\fontspec[BoldFont=texgyreheros-bold.otf]{texgyreheros-regular.otf} \sample}
\end{tabular}

\end{document}
```

**Reply by note286** [reply]
现在的问题就是规定和执行不符，遭罪的是学生，最好的办法就是派个代表沟通一下。
